server/test/unit/common/config/test_external_config.py

```diff
@@ -0,0 +1,107 @@
+                response = session.get(f"{server}/api/v0.2/config")
+                data_config = response.json()
+                self.assertEqual(data_config["config"]["displayNames"]["dataset"], "pbmc3k")
+                self.assertTrue(data_config["config"]["parameters"]["disable-diffexp"])
+
+            env["DATAPATH"] = f"{FIXTURES_ROOT}/a95c59b4-7f5d-4b80-ad53-a694834ca18b.h5ad"
+            env["DIFFEXP"] = "True"
+            with test_server(command_line_args=["-c", configfile], env=env) as server:
+                session = requests.Session()
+                response = session.get(f"{server}/api/v0.2/config")
+                data_config = response.json()
+                self.assertEqual(
+                    data_config["config"]["displayNames"]["dataset"], "a95c59b4-7f5d-4b80-ad53-a694834ca18b"
+                )
+                self.assertFalse(data_config["config"]["parameters"]["disable-diffexp"])
+
+    @patch("server.common.config.external_config.get_secret_key")
+    def test_aws_secrets_manager(self, mock_get_secret_key):
+        mock_get_secret_key.return_value = {
+            "oauth_client_secret": "mock_oauth_secret",
+            "db_uri": "mock_db_uri",
+        }
+        with tempfile.TemporaryDirectory() as tempdir:
+            configfile = os.path.join(tempdir, "config.yaml")
+            with open(configfile, "w") as fconfig:
+                config = """
+                external:
+                  aws_secrets_manager:
+                    region: us-west-2
+                    secrets:
+                      - name: my_secret
+                        values:
+                          - key: flask_secret_key
+                            path: [server, app, flask_secret_key]
+                            required: false
+                          - key: db_uri
+                            path: [dataset, user_annotations, hosted_tiledb_array, db_uri]
+                            required: true
+                          - key: oauth_client_secret
+                            path: [server, authentication, params_oauth, client_secret]
+                            required: true
+                """
+                fconfig.write(config)
+
+            app_config = AppConfig()
+            app_config.update_from_config_file(configfile)
+            app_config.server_config.single_dataset__datapath = f"{FIXTURES_ROOT}/pbmc3k.cxg"
+            app_config.server_config.app__flask_secret_key = "original"
+            app_config.server_config.single_dataset__datapath = f"{FIXTURES_ROOT}/pbmc3k.cxg"
+
+            app_config.complete_config()
+
+            self.assertEqual(app_config.server_config.app__flask_secret_key, "original")
+            self.assertEqual(app_config.server_config.authentication__params_oauth__client_secret, "mock_oauth_secret")
+            self.assertEqual(
+                app_config.default_dataset_config.user_annotations__hosted_tiledb_array__db_uri, "mock_db_uri"
+            )
```

Comment:
I think this test suite is missing a bunch of negative cases. There's a good chunk of logic in `ExternalConfig` around ensuring that wrong config errors are raised and I'd like to see test cases that cover those to ensure that we're not letting through configs that we expect to raise errors.